### PR TITLE
Migrate Imagination gpu 1.18 kernel module from linux 5.17 to 6.6

### PR DESCRIPTION
**Pull request by Sakura286**
Migrate Imagination gpu 1.18 kernel module from linux 5.17 to 6.6

Some changes are quite rough. Need someone help to review.

orig: https://github.com/RevySR/linux-eswin/pull/1

### CODE REVIEW

**Review by linmineswincomputing — CHANGES_REQUESTED** [review]
Do not change sifive_ccache.c,. Using arch_sync_dma_for_device/cpu instead of sifive_l2_flush64_range in  sysconfig.c

**Review by linmineswincomputing — CHANGES_REQUESTED** [review]
#include <linux/dma-map-ops.h> to use arch_sync_dma_for_cpu/device

**Comment by linmineswincomputing** [comment]
It would be better to combine these commits into one commit, since these commits are the  same topic. It will make the kernel commit history more clean.

File: drivers/soc/sifive/sifive_ccache.c
Comment (linmineswincomputing) [inline drivers/soc/sifive/sifive_ccache.c:1]: Do not change sifive_ccace.c. Other drivers shhould  use arch_sync_dma_for_device/cpu.

File: include/soc/sifive/sifive_ccache.h
Comment (linmineswincomputing) [inline include/soc/sifive/sifive_ccache.h:1]: Do not change sifive_ccace.h. Other drivers shhould use arch_sync_dma_for_device/cpu

**Comment by Sakura286** [comment]
> It would be better to combine these commits into one commit, since these commits are the same topic. It will make the kernel commit history more clean.

Splitting the commits might be more friendly for reviewer. It seems that the reviewer can squash or rebase multiple commits?

File: drivers/soc/sifive/sifive_ccache.c
Comment (Sakura286) [reply drivers/soc/sifive/sifive_ccache.c:1]: Fixed. Please see https://github.com/eswincomputing/linux-stable/pull/1/commits/8b82c25c272d2ca026427c89afa83c9021af74d4

**Review by linmineswincomputing — APPROVED** [review]

**Pull request closed without merging**